trestle/core/catalog_interface.py

```diff
@@ -650,85 +651,200 @@ def write_catalog_as_markdown(self, context: ControlContext, part_id_map: Dict[s
-                        new_dict[const.DISPLAY_NAME] = display_name
-                    set_param_dict[param_id] = new_dict
-                if set_param_dict:
-                    if const.SET_PARAMS_TAG not in new_context.yaml_header:
-                        new_context.yaml_header[const.SET_PARAMS_TAG] = {}
-                    if new_context.overwrite_header_values:
-                        # update the control params with new values
-                        for key, value in new_context.yaml_header[const.SET_PARAMS_TAG].items():
-                            if key in control_param_dict:
-                                set_param_dict[key] = value
-                    else:
-                        # update the control params with any values in yaml header not set in control
-                        # need to maintain order in the set_param_dict
-                        for key, value in new_context.yaml_header[const.SET_PARAMS_TAG].items():
-                            if key in control_param_dict and key not in set_param_dict:
-                                set_param_dict[key] = value
-                    new_context.yaml_header[const.SET_PARAMS_TAG] = set_param_dict
-                elif const.SET_PARAMS_TAG in new_context.yaml_header:
-                    # need to cull any params that are not in control
-                    pop_list: List[str] = []
-                    for key in new_context.yaml_header[const.SET_PARAMS_TAG].keys():
-                        if key not in control_param_dict:
-                            pop_list.append(key)
-                    for pop in pop_list:
-                        new_context.yaml_header[const.SET_PARAMS_TAG].pop(pop)
-            _, group_title, _ = self.get_group_info_by_control(control.id)
-            # control could be in sub-group of group so build path to it
-            group_dir = new_context.md_root
-            control_path = self.get_control_path(control.id)
-            for sub_dir in control_path:
-                group_dir = group_dir / sub_dir
-                if not group_dir.exists():
-                    group_dir.mkdir(parents=True, exist_ok=True)
+                for set_param in as_list(new_context.control_implementation.set_parameters):
+                    _update_values(set_param)
+                for imp_req in as_list(new_context.control_implementation.implemented_requirements):
+                    if imp_req.control_id == control.id:
+                        for set_param in as_list(imp_req.set_parameters):
+                            _update_values(set_param)
+
+                if control_param_dict:
+                    new_context.yaml_header[const.PARAM_VALUES_TAG] = {}
+                    for key, param in control_param_dict.items():
+                        new_context.yaml_header[const.PARAM_VALUES_TAG][key] = none_if_empty(
+                            ControlInterface.param_to_str(param, ParameterRep.VALUE_OR_EMPTY_STRING)
```

Comment:
This might fail as param can be a string now (after _update_values)

Reply:
it is now param